flink-connectors/flink-connector-rabbitmq/src/test/java/org/apache/flink/streaming/connectors/rabbitmq/RMQSourceTest.java

```diff
@@ -419,6 +421,40 @@ protected Connection setupConnection() throws Exception {
         Mockito.verify(channel, Mockito.times(0)).basicQos(anyInt());
     }
 
+    private static class CallsRealMethodsWithDelay extends CallsRealMethods {
+
+        private final long delay;
+
+        public CallsRealMethodsWithDelay(long delay) {
+            this.delay = delay;
+        }
+
+        public Object answer(InvocationOnMock invocation) throws Throwable {
+            Thread.sleep(delay);
+            return super.answer(invocation);
+        }
+    }
+
+    @Test(timeout = 10000L)
+    public void testDeliveryTimeout() throws Exception {
+        source.autoAck = false;
+        // mock delivery delay
+        Mockito.when(source.consumer.nextDelivery())
+                .then(new CallsRealMethodsWithDelay(15000L))
+                .thenThrow(new RuntimeException());
+        Mockito.when(source.consumer.nextDelivery(any(Long.class)))
+                .then(new CallsRealMethodsWithDelay(10L))
+                .thenReturn(null);
+        sourceThread.start();
+        // wait a bit for the source to start
+        Thread.sleep(5);
```

Comment:
In general, I'd recommend not relying on time in test cases because it may take longer on other systems to start the source which may introduce a test instability.

Reply:
You're right. This would require extending the test into:
1) generating and consuming at least 1 message (we need to make sure that the consumer was started), instead of waiting arbitrary 5ms 
2) stopping the message delivery (similarly as its currently implemented, but after the source was started)
3) cancelling the source and waiting for it to stop (as its currently implemented)
4) checking if the job stopped gracefully (this can be verified by adding some helper boolean flag) instead of checking the exception (so this would eliminate the second Thread.sleep).

Reply:
I refactored the delivery timeout unit test so that it does not rely on any timeouts or processing time at all (changes in a229e25). I also moved the helper class after all `@Test` methods.